youths/schema.py

```diff
@@ -0,0 +1,194 @@
+
+
+class UpdateYouthProfile(graphene.Mutation):
+    class Arguments:
+        youth_profile_data = UpdateYouthProfileInput(required=True)
+        profile_id = graphene.Int()
+
+    youth_profile = graphene.Field(YouthProfileType)
+
+    @login_required
+    def mutate(self, info, **kwargs):
+        input_data = kwargs.pop("youth_profile_data")
+        resend_request_notification = input_data.pop(
+            "resend_request_notification", False
+        )
+        if (
+            info.context.user.is_superuser
+        ):  # FIXME should be something like user.is_staff_or_more?
+            profile = Profile.objects.get(pk=kwargs.pop("profile_id"))
+        else:
+            profile = Profile.objects.get(user=info.context.user)
+        youth_profile = YouthProfile.objects.get(profile=profile)
+        for field, value in input_data.items():
+            setattr(youth_profile, field, value)
+        youth_profile.save()
+
+        if resend_request_notification:
+            youth_profile.approval_token = create_approval_token()
+            mail_encrypted_approval_link(youth_profile)
+            youth_profile.approval_notification_sent = datetime.now()
+            youth_profile.save()
+
+        return UpdateYouthProfile(youth_profile=youth_profile)
+
+
+class ApproveYouthProfileInput(graphene.InputObjectType):
+    diabetes = graphene.Boolean()
+    epilepsy = graphene.Boolean()
+    heart_disease = graphene.Boolean()
+    extra_illnesses_info = graphene.String()
+    serious_allergies = graphene.Boolean()
+    allergies = graphene.String()
+    photo_usage_approved = graphene.Boolean()
+
+
+class ApproveYouthProfile(graphene.Mutation):
+    class Arguments:
+        approval_token = graphene.String(required=True)
+        approval_data = ApproveYouthProfileInput(required=True)
+
+    youth_profile = graphene.Field(YouthProfileType)
+
+    def mutate(self, info, **kwargs):
+        youth_data = kwargs.pop("approval_data")
+        token = kwargs.get("approval_token")
+
+        try:
+            data = signing.loads(token)
+            youth_profile = YouthProfile.objects.get(
+                pk=data.get("id"), approval_token=data.get("approval_token")
+            )
+        except (ObjectDoesNotExist, signing.BadSignature):
+            raise Exception("Invalid approval token")
+
+        for field, value in youth_data.items():
+            setattr(youth_profile, field, value)
+
+        youth_profile.approved_time = datetime.now()
+        youth_profile.approval_token = ""  # invalidate
+        youth_profile.save()
+        send_notification(
+            email=youth_profile.profile.email,
+            notification_type="youth_registration_approved",
+            context={"youth_profile": youth_profile},
+        )
+        return ApproveYouthProfile(youth_profile=youth_profile)
+
+
+class Query(graphene.ObjectType):
+    youth_profile = graphene.Field(YouthProfileType, profile_id=graphene.Int())
```

Comment:
graphene has `ID` scalar, which can be and `Int` or `String`, so `profile_id=graphene.ID()` would be more appropriate